CMakeLists.txt

```diff
@@ -436,10 +436,17 @@ configure_file(
   ${CMAKE_CURRENT_BINARY_DIR}/libfolly.pc.gen
   @ONLY
 )
+
+# Specify target to allow resolving generator expressions requiring a target for CMake >=3.19. See #1414.
+if(CMAKE_VERSION VERSION_GREATER 3.18)
```

Comment:
I also replaced `VERSION_GREATER_EQUAL` here as it's only available in CMake >=3.7.

Reply:
`VERSION_GREATER 3.18` matches on non-zero 3.18 patch versions as well, so it breaks with v3.18.2, for example.

Reply:
See #1439.